lib/ansible/modules/storage/glusterfs/gluster_heal_facts.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Red Hat, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: gluster_heal_facts
+short_description: Gather facts about self-heal or rebalance status
+author: "Devyani Kota (@devyanikota)"
+version_added: "2.8"
+description:
+  - Gather facts about either self-heal or rebalance status.
+options:
+  name:
+    description:
+      - The volume name.
+    required: true
+    aliases: ['volume']
+  status_filter:
+    default: "self-heal"
+    choices: ["self-heal", "rebalance"]
+    description:
+      - Determines which facts are to be returned. If the status_filter is
+        self-heal, status of self-heal, along with the number of files still
+        in process are returned. If status_filter is rebalance,
+        rebalance status is returned.
+requirements:
+  - GlusterFS > 3.2
+'''
+
+EXAMPLES = '''
+# Gather facts about all gluster hosts in the cluster and return self-heal
+# status along with the number of files still in process:
+- gluster_heal_facts:
+    name: test_volume
+    status_filter: self-heal
+- debug:
+    var: gluster_hosts
+'''
+
+RETURN = '''
+name:
+    description: GlusterFS volume name
+    returned: always
+    type: string
+status_filter:
+    description: Whether self-heal or rebalance status is to be returned
+    returned: always
+    type: string
+heal_info:
+    description: List of files that still need healing process
+    returned: On success
+    type: string
+rebalance_status:
+    description: Status of rebalance operation
+    returned: On success
+    type: string
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from distutils.version import LooseVersion
+
+glusterbin = ''
+
+
+def run_gluster(gargs, **kwargs):
+    global glusterbin
+    global module
+    args = [glusterbin, '--mode=script']
+    args.extend(gargs)
+    try:
+        rc, out, err = module.run_command(args, **kwargs)
+        if rc != 0:
+            module.fail_json(msg='error running gluster (%s) command (rc=%d): %s' %
+                             (' '.join(args), rc, out or err), exception=traceback.format_exc())
+    except Exception as e:
+        module.fail_json(msg='error running gluster (%s) command: %s' % (' '.join(args),
+                                                                         to_native(e)), exception=traceback.format_exc())
+    return out
+
+
+def get_self_heal_status(name):
+    out = run_gluster(['volume', 'heal', name, 'info'])
```

Comment:
Since you're screenscraping, set the locale to C before scraping.

See: gluster_peer